crates/extensions/tedge_downloader_ext/src/actor.rs

```diff
@@ -0,0 +1,93 @@
+use async_trait::async_trait;
+use download::Auth;
+use download::DownloadError;
+use download::DownloadInfo;
+use download::Downloader;
+use log::info;
+use std::path::Path;
+use std::path::PathBuf;
+use tedge_actors::Actor;
+use tedge_actors::ChannelError;
+use tedge_actors::MessageReceiver;
+use tedge_actors::RuntimeError;
+use tedge_actors::Sender;
+use tedge_actors::SimpleMessageBox;
+use tedge_utils::file::PermissionEntry;
+
+#[derive(Debug, Clone)]
+pub struct DownloadRequest {
+    pub url: String,
+    pub file_path: PathBuf,
+    pub auth: Option<Auth>,
+}
+
+impl DownloadRequest {
+    pub fn new(url: &str, file_path: &Path, auth: Option<Auth>) -> Self {
+        Self {
+            url: url.into(),
+            file_path: file_path.into(),
+            auth,
+        }
+    }
+}
+
+pub type DownloadResult = Result<PathBuf, DownloadError>;
+
+#[derive(Debug)]
+pub struct DownloadResponse {
+    pub result: DownloadResult,
+}
```

Comment:
Something is up-side down here. What you want to be able to extend in not the fact we can have errors (i.e adding things along the result), but what is returned in case things are okay. Today you return only the path, but later your might want to add the url (by the way I would add this url).

```suggestion
pub type DownloadResult = Result<DownloadResponse, DownloadError>;

#[derive(Debug)]
pub struct DownloadResponse {
        pub url: String,
        pub file_path: PathBuf,
}
```